### PR TITLE
Add more cfg aliases

### LINKED ISSUE #2188: Tracking Issue: Replace big `cfg!` chunks with shorter aliases
## Background

With the crate [`cfg_aliases`](https://crates.io/crates/cfg_aliases), we can replace common, chunky `cfg!`s with shorter aliases, here are the aliases defined by us:

* apple_targets: `any(ios, macos, watchos, tvos)`
* freebsdlike: `any(freebsd, dragonfly)`
* bsd: `any(freebsd, dragonfly, netbsd, openbsd, apple_targets)`
* linux_android: `any(linux, android)`
* solarish: `any(illumos, solaris)`

#2205 has done most of the replacements, but there are still `cfg!`s that haven't been updated, we want to clear them all.

## TODO

Scan our source code, and replace the long `cfg!` chunks with shorter aliases:

* [x] `src/dir.rs` `src/env.rs` (#2214)
* [x] `src/errno.rs` (#2214)
* [ ] `src/fcntl.rs` (#2215)
* [ ] `src/features.rs` `src/ifaddrs.rs` (#2215)
* [ ] `src/kmod.rs` `src/lib.rs` (#2215)
* [ ] `src/macro.rs` (#2215)
* [ ] `src/mqueue.rs` (#2215)
* [ ] `src/poll.rs` (#2215)
* [ ] `src/pty.rs` (#2215)
* [ ] `src/sched.rs` (#2215)
* [ ] `src/time.rs` (#2215)
* [x] `src/unistd.rs` (#2216)
* [x] `src/mount/bsd.rs` (#2217)
* [x] `src/mount/linux.rs` `src/mount/mod.rs` (#2217)
* [x] `src/net/if_.rs` (#2217)
* [x] `src/sys/aio.rs` (#2218)
* [x] `src/sys/epoll.rs` (#2218)
* [x] `src/sys/event.rs` (#2218)
* [x] `src/sys/mman.rs` (#2218)
* [x] `src/sys/mod.rs` (#2218)
* [x] `src/sys/resource.rs` (#2218)
* [x] `src/sys/sendfile.rs` (#2218)
* [x] `src/sys/signal.rs` (#2218)
* [x] `src/sys/stat.rs` (#2219)
* [x] `src/sys/statfs.rs` (#2219)
* [x] `src/sys/termios.rs` (#2220)
* [ ] `src/sys/time.rs` (#2229)
* [x] `src/sys/uio.rs` `src/sys/wait.rs` (#2221)
* [x] `src/sys/ioctl/mod.rs` (#2222)
* [x] `src/sys/ptrace/mod.rs` `src/sys/ptrace/bsd.rs` (#2222)
* [x] `src/sys/socket/addr.rs` (#2223)
* [x] `src/sys/socket/mod.rs` (#2224)
* [x] `src/sys/socket/sockopt.rs` (#2225)
* [ ] `test/common/mod.rs` (#2230)
* [ ] `test/sys/mod.rs` `test/sys/test_socket.rs` (#2230)
* [ ] `test/sys/test_sockopt.rs` (#2230)
* [ ] `test/test.rs` (#2230)
* [ ] `test/test_fcntl.rs` (#2230)
* [ ] `test/test_net.rs` `test/test_poll.rs` (#2230)
* [ ] `test/test_sendfile.rs` (#2230)
* [ ] `test/test_stat.rs` (#2230)
* [ ] `test/test_time.rs` (#2230)
* [ ] `test/test_unistd.rs` (#2230)

### PR DESCRIPTION
**Pull request by asomers**
Add cfg aliases for linux_android, bsd, and freebsdlike.  Use them in many places, though not everywhere they could theoretically be used.

Fixes #2188

## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [ ] I have written necessary tests and rustdoc comments
- [ ] A change log has been added if this PR modifies nix's API

### CODE REVIEW

**Comment by SteveLauC** [comment]
Should we also do this for Solaris and Illumos?

```rs
cfg_aliases! {
    illumos: { target_os = "illumos" },
    solaris: { target_os = "solaris" },
        
    solarish: { any(solaris, illumos) }
}
```


BTW, we should document this in probably `CONVENTIONS.md` (or `CONTRIBUTING.md`?) so that our contributors can be aware of these aliases and use them in PRs

File: build.rs [hunk 1]
```diff
@@ -2,10 +2,19 @@ use cfg_aliases::cfg_aliases;
...
         ios: { target_os = "ios" },
+        freebsd: { target_os = "freebsd" },
+        linux: { target_os = "linux" },
         macos: { target_os = "macos" },
+        netbsd: { target_os = "netbsd" },
+        openbsd: { target_os = "openbsd" },
         watchos: { target_os = "watchos" },
         tvos: { target_os = "tvos" },
         apple_targets: { any(ios, macos, watchos, tvos) },
+        bsd: { any(freebsd, dragonfly, ios, macos, netbsd, openbsd, tvos, watchos) },
```
Comment (SteveLauC) [inline build.rs:16]: Might be better to use the defined `apple_targets` here:

```suggestion
        bsd: { any(freebsd, dragonfly, netbsd, openbsd, apple_targets) },
```

File: build.rs [hunk 2]
```diff
@@ -2,10 +2,19 @@ use cfg_aliases::cfg_aliases;
...
+        linux_android: { any(android, linux) },
+        freebsdlike: { any(dragonfly, freebsd)}
```
Comment (SteveLauC) [inline build.rs:18]:
```suggestion
        freebsdlike: { any(dragonfly, freebsd) }
```

**Comment by Jan561** [comment]
Are other config-aliases, like an alias `mmsg` for
```rust
#[cfg(all(
    feature = "socket",
    feature = "uio",
    any(
        target_os = "linux",
        target_os = "android",
        target_os = "freebsd",
        target_os = "netbsd",
    ),
))]
```
desirable for the future? Or should they just stay platform-related for the time being?

**Comment by asomers** [comment]
> Are other config-aliases, like an alias `mmsg` for
> 
> ```rust
> #[cfg(all(
>     feature = "socket",
>     feature = "uio",
>     any(
>         target_os = "linux",
>         target_os = "android",
>         target_os = "freebsd",
>         target_os = "netbsd",
>     ),
> ))]
> ```
> 
> desirable for the future? Or should they just stay platform-related for the time being?

I think we can add them on an as-needed basis.

**Review by SteveLauC — CHANGES_REQUESTED** [review]

**Comment by SteveLauC** [comment]
I just checked that the changes made in this PR are good.

But there are also still `cfg`s that haven't been updated, we should handle them as well, but not in this PR, instead, we should probably split the remaining job into several PRs, which will make code review easier.

For #2188, let's turn it into a tracking issue, once the tasks in it are all done, then we can close it.

----------

For the requested changes, I have committed 2 with the GitHub UI, please ignore the left ones and handle them in other PRs

**Pull request merged**